src/vma/sock/sockinfo.cpp

```diff
@@ -351,6 +405,23 @@ int sockinfo::setsockopt(int __level, int __optname, const void *__optval, sockl
 				si_logdbg("SOL_SOCKET, %s=\"???\" - NOT HANDLED, optval == NULL", setsockopt_so_opt_to_str(__optname));
 			}
 			break;
+		case SO_VMA_RING_ALLOC_LOGIC:
+			if (__optval) {
+				if (__optlen == sizeof(vma_ring_alloc_logic_attr)) {
+					vma_ring_alloc_logic_attr *attr = (vma_ring_alloc_logic_attr *)__optval;
+					return set_ring_attr(attr);
+				}
+				else {
+					si_logdbg("SOL_SOCKET, %s=\"???\" - bad length expected %d got %d",
+						  setsockopt_so_opt_to_str(__optname),
+						  sizeof(vma_ring_alloc_logic_attr), __optlen);
+					break;
+				}
+			}
+			else {
+				si_logdbg("SOL_SOCKET, %s=\"???\" - NOT HANDLED, optval == NULL", setsockopt_so_opt_to_str(__optname));
+			}
+			break;
```

Comment:
Please use the following return values :

```
/*
 * Sockinfo setsockopt() return values
 */
#define	SOCKOPT_INTERNAL_VMA_SUPPORT  0    // Internal socket option, should not pass request to OS.
#define	SOCKOPT_NO_VMA_SUPPORT       -1    // Socket option was found but not supported, error should be returned to user.
#define	SOCKOPT_PASS_TO_OS            1	   // Should pass to TCP/UDP level or OS.
```

Reply:
Done